README.md
# Description

This cookbook installs PHPIPAM on an Ubuntu 14.04 system.

It can be used to install a standalone app/db multi-role system on a single
system, or configured to install the application and database components on
separate servers. It also has a recipe to install a job to backup the PHPIPAM
database (see the below section).

NOTE: only specific versions of PHPIPAM are supported. We start with version
1.2, and will probably be adding 1.3 along with upgrade options eventually.

You can view all the PHPIPAM documentation at https://phpipam.net/.

## About Backups

Backups are installed and performed using the [Ruby backup gem][1] if the
`phpipam::backup` recipe is included in the run list. By default, backups are
made locally and kept for 7 days, but this is configurable. S3 options are also
available. See the attributes and recipe sections for more details.

[1]: https://github.com/backup/backup

### Using Custom Notifiers for Backups

Notifiers can be configured for backups. For the exact syntax that you want to
use, see the details for the backup gem's [Notifier resource][2]. The notifiers
are supplied to the `node['phpipam']['backup_notifiers']` attribute, as an array
of hashes, with the `type:` key specifying the notifier type, and the rest of
the options specified as a hash to the `opts:` key, like below. Note that by
default, notifiers are configured to notify on success, warning, and failure.

[2]: http://backup.github.io/backup/v4/notifiers/

```
node.default['phpipam']['backup_notifiers'] = [
  {
    type: 'Mail',
    opts: {
      'from' => 'nobody@example.com',
      'to' => 'somebody@example.com',
    }
  }
]
```

### Ignoring Certain Backup Warnings

You can also ignore certain backup warnings via
`node['pbpipam']['backup_ignore_warning_messages']`:

```
node.default['pbpipam']['backup_ignore_warning_messages'] = [
  /fog: the specified s3 bucket name\(.*\) contains a/,
  'some-exact-string-match'
]
```

**NOTE:** If you want to use literals more complex than a string (like the regex
above), you need to specify your node attributes with Ruby - either use a
wrapper cookbook or a process that passes your attributes in Ruby. Try to avoid
JSON or YAML. **Note that this behaviour is not necessarily Chef compliant** -
for example, your `Regexp` literals will show up as strings when saved to the
Chef server. You have been warned!

## About Releases

This cookbook has not been released on Supermarket. This may change in the
future, but until then, you need to take extra steps to include it in your
Chef server or other cookbooks with Berkshelf.

### Including in Berkshelf

Releases are tagged, so you can include the cookbook at a certain release simple
enough. Below is an example `Berksfile`. Change the release version below to
the release that you want.

```
source "https://supermarket.chef.io"

cookbook 'phpipam', git: 'https://github.com/paybyphone/phpipam_cookbook.git', tag: 'v9.9.99'

metadata
```

### Adding Direct to Chef Server

To add this cookbook to a Chef server directly, clone the repository, check out
the release you want, and then run the Berkshelf commands to add it:

```
git clone https://github.com/paybyphone/phpipam_cookbook.git
cd phpipam_cookbook
git checkout v9.9.99
berks install && berks upload
```

# Requirements

## Platform:

* ubuntu (= 14.04)

## Cookbooks:

* apache2 (~> 3.2)
* apt (~> 6.0)
* build-essential (~> 8.0)
* database (~> 6.1)
* mysql (~> 8.2)
* php (~> 2.2)
* poise-ruby (~> 2.2)

# Attributes

* `node['phpipam']['db_host']` - The host name to connect to for the PHPIPAM database. Defaults to `localhost`.
* `node['phpipam']['db_name']` - The name of the PHPIPAM database. Defaults to `phpipam`.
* `node['phpipam']['db_user']` - The user name for the PHPIPAM database. Defaults to `phpipam`.
* `node['phpipam']['db_password']` - The password for the created PHPIPAM database. Defaults to `phpipamadmin`.
* `node['phpipam']['db_root_password']` - The root password for the MySQL installation of the PHPIPAM database. Defaults to `atotallyrandompassword`.
* `node['phpipam']['docroot']` - The document root, or where to install the PHPIPAM content. Defaults to `/var/www/html/phpipam`.
* `node['phpipam']['apache2_keep_default_vhost']` - Keep the default virtualhost on Apache 2 installations.

You may want to set this to true on a system where PHPIPAM is not the only
site. Leaving this at the default will ensrue that the virtual host installed
is the default virtual host on the system. Defaults to `false`.
* `node['phpipam']['uri_base']` - The URI base to install PHPIPAM to. This will mean that the GUI and API will
be accessible through http(s)://example.com/phpipam/, for example, if this
was set to "/phpipam/". Defaults to `/`.
* `node['phpipam']['vhost_name']` - The virtual host for the PHPIPAM site. Defaults to `nil`.
* `node['phpipam']['vhost_aliases']` - Any aliases that should be used to access PHPIPAM. Defaults to `nil`.
* `node['phpipam']['version']` - The version of IPAM to fetch the content for (and ultimately install).

This can currently be one of `1.2` and `dev`. `dev` is an alias to a current
commit snapshotted off of master, with schema and configuration snapshotted off
of that commit. See `recipes/app_site.rb` for the specific commit dev tracks. Defaults to `1.2`.
* `node['phpipam']['download_path']` - The location to download the PHPIPAM content to. This is set to a directory
in /tmp to start with, and is overridden to the document root when running
the web content recipes. Defaults to `/tmp/phpipam`.
* `node['phpipam']['content_owner']` - The owner of the downloaded content. This is modified to the system's web
user for web installations. Defaults to `root`.
* `node['phpipam']['content_group']` - The group of the downloaded content. This is modified to the system's web
group for web installations. Defaults to `root`.
* `node['phpipam']['install_mysql']` - Install MySQL as part of the cookbook deployment. Defaults to `true`.
* `node['phpipam']['backup_user']` - The user to create to perform backups under. A group is also created under
this user name. Defaults to `ipambackup`.
* `node['phpipam']['backup_home']` - The home directory for backup configuration data and content. Defaults to `/var/ipambackup`.
* `node['phpipam']['backup_tmp_path']` - The directory to hold backup data while performing backups. Defaults to `#{default['phpipam']['backup_home']}/tmp`.
* `node['phpipam']['backup_enable_logfile']` - Enable logging backups to a log file. These logs go to the "log" directory
in the directory defined by `node['phpipam']['backup_home']`. Defaults to `true`.
* `node['phpipam']['backup_enable_syslog']` - Enable syslog logging. Logs get logged as `backup` under facility local0. Defaults to `false`.
* `node['phpipam']['backup_storage_path']` - The directory to store locally-kept backups. Defaults to `#{default['phpipam']['backup_home']}/storage`.
* `node['phpipam']['backup_s3_enabled']` - Enable S3 backup storage. Defaults to `false`.
* `node['phpipam']['backup_s3_access_key_id']` - The AWS Access Key ID for S3 backup storage. Defaults to ``.
* `node['phpipam']['backup_s3_secret_access_key']` - The AWS secret access key for S3 backup storage. Defaults to ``.
* `node['phpipam']['backup_s3_region']` - The AWS region your bucket is in. Defaults to ``.
* `node['phpipam']['backup_s3_bucket']` - The S3 bucket name for S3 storage. Defaults to ``.
* `node['phpipam']['backup_s3_path']` - The key (path) S3 backups will be stored in. Defaults to ``.
* `node['phpipam']['backup_keep_days_local']` - The number of days to keep local backups. Defaults to `7`.
* `node['phpipam']['backup_keep_days_s3']` - The number of days to keep S3 backups. Defaults to `90`.
* `node['phpipam']['backup_notifiers']` -  Defaults to `[ ... ]`.
* `node['phpipam']['backup_cron_minute']` - The minute to run the backup job on (0-59). Defaults to `00`.
* `node['phpipam']['backup_cron_hour']` - The hour to run the backup job on (0-23). Defaults to `02`.
* `node['phpipam']['backup_cron_weekday']` - The weekday to run the backup job on (0-6, Sun-Sat, or * for every day). Defaults to `*`.
* `node['phpipam']['backup_ignore_warning_messages']` - A list of warning messages to ignore. Defaults to `[ ... ]`.

# Recipes

* [phpipam::app_db](#phpipamapp_db) - This recipe installs the PHPIPAM database by creating the DB (root access required to the DB server) and importing the schema.
* [phpipam::app_site](#phpipamapp_site) - This recipe installs and configures the application and its content.
* [phpipam::backup](#phpipambackup) - This recipe installs backups using the [Ruby backup gem][1].

## phpipam::app_db

This recipe installs the PHPIPAM database by creating the DB (root access
required to the DB server) and importing the schema. It also installs MySQL,
if `node['phpipam']['install_mysql']` is set to true (the default).

## phpipam::app_site

This recipe installs and configures the application and its content. It also
installs Apache 2, if not installed already.

## phpipam::backup

This recipe installs backups using the [Ruby backup gem][1]. Options are
available in node attributes for sending the backups to S3, syslog logging, and
specifying notifier configuration through the
`node['phpipam']['backup_notifiers']` array. See the description section for
examples.

[1]: https://github.com/backup/backup


# License and Maintainer

Maintainer:: PayByPhone (<systems@paybyphone.com>)

License:: Apache 2.0


CHANGELOG.md
## 0.4.0

Dev branch has been updated to a version that supports nested custom fields -
see the PR and issue referenced in v0.3.0 for more details.

## 0.3.0

This release adds support for tracking a `dev` version of PHPIPAM. This is a
commit off of master, that will hopefully end up being version 1.3 at some point
in time.

This version also currently tracks PBP's fork at
https://github.com/paybyphone/phpipam while the nested custom field issues
outlined in https://github.com/phpipam/phpipam/issues/1073 and
https://github.com/phpipam/phpipam/pull/1100 are fixed.

## 0.2.2

 * `uri_base` now has its trailing slash, if present, trimmed on non-default
   setups. Docroot path is also normalized to ensure that trailing slashes are
   present everywhere in the apache config. This is to support proxies where
   the bare word URI (like `/phpipam`) is desired and the user does not want to
   set up an extra redirect.

## 0.2.1

This update moves S3 storage to the top of the storage order in the backups
recipe. This ensures that if S3 is included, that backups are properly moved to
local storage versus copied, avoiding extra I/O  and storage issues, in addition
to keeping unnecessary warnings from happening.

## 0.2.0

**NOTE:** To avoid bugs and errors, do not use any 0.1.x version of this
cookbook - there may be certain features that were not usable as intended.

Fixes since 0.1.8:

 * Fixed backup cron job, path to job now uses an absolute path. This ensures
   that the job runs.
 * Ensure that the default `/var/run/mysql` symlink that we create does not get
   blown away on subsequent Chef runs other than the first one.
 * Change the backup user to `ipambackup`, and the default home directory to
   `/var/ipambackup`. This ensures that the installed backup job does not
   interfere with any distro-level backup processes that use the default
   `backup` user and `/var/backup` directory.

## 0.1.8

Added the `backup_ignore_warning_messages` node attribute which allows you
to configure warning messages to ignore.

Also fixed a small indentation issue on the backup model.

## 0.1.7

Fixed the backup model template so that it doesn't render the default
`on_success`, `on_warning`, or `on_failure` options if user-defined ones are
given.

## 0.1.6

This update adds backup support - installed using the ruby backup gem.
Backup options can be set up to use S3 in addition to local storage
(mandatory) with configurable thresholds for each. You can also
configure a custom number of notifiers.

## 0.1.5
    
The `uri_base` option is back. Defining a `uri_base` other than `/` does the following:

 * Sets `BASE` and `RewriteBase` in `config.php` and `.htaccess` respectively to
   whatever you define
 * Adds an `Alias` directive in Apache for the correct URI path (in case
   you are not using this to reverse proxy)

Also explicitly forced the MPM in use to prefork.

## 0.1.4

Added `build-essential` to the cookbook chain, as building the `mysql2` gem
requires `make`, which may not be available on minimal systems.

## 0.1.3

Added version automation off of `metadata.rb`. The cookbook is now versioned off
of this `CHANGELOG.md` file. To release, you need to switch to the release tag
that you want to release or upload, and then do your needful (example: `berks
install && berks upload`, or cookbook release if we so add it later).

Using versions prior to this one are not recommended as they will be stuck at
version `0.1.0`.

## 0.1.2

Bump a version to lock in documentation updates.

## 0.1.1

This release should be used over `0.1.0` as there were some bugs that make using
it unusable.

 * Removed the `uri_base` attribute as it was never usable and I've decided to
   not implement it.
 * Remove the `install_apache` option as the Apache virtual host needs to be
   dropped in for this to be fully functional. Coexistence with other virtual
   hosts on the same server is still possible if `apache2_keep_default_vhost` is
   set.

## 0.1.0

Initial release.


test/cookbooks/backup_defaults/metadata.rb
name 'backup_defaults'


test/cookbooks/backup_defaults/recipes/default.rb
node.default['phpipam']['backup_ignore_warning_messages'] = [
  /this is a test regex/,
  'this is a string match'
]


test/cookbooks/phpipam_sdk_go_testacc/metadata.rb
name 'phpipam_sdk_go_testacc'


recipes/app_db.rb
=begin
#<
This recipe installs the PHPIPAM database by creating the DB (root access
required to the DB server) and importing the schema. It also installs MySQL,
if `node['phpipam']['install_mysql']` is set to true (the default).
#>
=end

include_recipe 'phpipam::_check_ipam_version'
include_recipe 'apt::default'
include_recipe 'build-essential::default'
include_recipe 'phpipam::_install_mysql' if node['phpipam']['install_mysql']

cookbook = run_context.cookbook_collection[cookbook_name]
schema_cache_path = cookbook.preferred_filename_on_disk_location(
  run_context.node,
  :files,
  "ipam_schema_#{node['phpipam']['version']}.sql"
)

TABLE_COUNT_SQL = <<EOS.freeze
mysql -h localhost -u#{node['phpipam']['db_user']} \
-p#{node['phpipam']['db_password']} #{node['phpipam']['db_name']} -B -N \
-e "select count(*) from information_schema.tables \
  where table_type = 'BASE TABLE' \
  and table_schema = '#{node['phpipam']['db_name']}'" | egrep '^0$'
EOS

mysql_client 'default' do
  action :create
end

chef_gem 'mysql2' do
  compile_time false
  action :install
end

mysql_connection_info = {
  host:     node['phpipam']['db_host'],
  username: 'root',
  password: node['phpipam']['db_root_password']
}

mysql_database node['phpipam']['db_name'] do
  connection mysql_connection_info
  action :create
end

mysql_database_user node['phpipam']['db_user'] do
  connection mysql_connection_info
  password node['phpipam']['db_password']
  database_name node['phpipam']['db_name']
  host 'localhost'
  action :grant
end

execute 'import_ipam_schema' do
  command <<EOS
mysql -h localhost -u#{node['phpipam']['db_user']} \
-p#{node['phpipam']['db_password']} #{node['phpipam']['db_name']} \
< #{schema_cache_path}
EOS
  action :run
  only_if TABLE_COUNT_SQL
end


recipes/backup.rb
=begin
#<
This recipe installs backups using the [Ruby backup gem][1]. Options are
available in node attributes for sending the backups to S3, syslog logging, and
specifying notifier configuration through the
`node['phpipam']['backup_notifiers']` array. See the description section for
examples.

[1]: https://github.com/backup/backup

#>
=end

package 'zlib1g-dev' do
  action :install
end

ruby_runtime '2'

ruby_gem 'backup'

group node['phpipam']['backup_user'] do
  action :create
end

user node['phpipam']['backup_user'] do
  action :create
  gid node['phpipam']['backup_user']
  home node['phpipam']['backup_home']
  manage_home true
end

[
  "#{node['phpipam']['backup_home']}/etc",
  "#{node['phpipam']['backup_home']}/etc/models",
  "#{node['phpipam']['backup_home']}/log",
  "#{node['phpipam']['backup_home']}/data",
  "#{node['phpipam']['backup_tmp_path']}",
  "#{node['phpipam']['backup_storage_path']}"
].each do |dir|
  directory dir do
    user node['phpipam']['backup_user']
    group node['phpipam']['backup_user']
    action :create
  end
end

template "#{node['phpipam']['backup_home']}/etc/config.rb" do
  action :create
  group node['phpipam']['backup_user']
  source 'backup_config.rb.erb'
  user node['phpipam']['backup_user']
  variables(
    backup_home:              node['phpipam']['backup_home'],
    backup_tmp_path:          node['phpipam']['backup_tmp_path'],
    backup_enable_logfile:    node['phpipam']['backup_enable_logfile'],
    backup_enable_syslog:     node['phpipam']['backup_enable_syslog'],
    ignore_warning_messages:  node['phpipam']['backup_ignore_warning_messages']
  )
end

template "#{node['phpipam']['backup_home']}/etc/models/phpipam_backup.rb" do
  action :create
  group node['phpipam']['backup_user']
  source 'backup_model.rb.erb'
  user node['phpipam']['backup_user']
  variables(
    db_name:                      node['phpipam']['db_name'],
    db_username:                  node['phpipam']['db_user'],
    db_password:                  node['phpipam']['db_password'],
    db_host:                      node['phpipam']['db_host'],
    backup_storage_path:          node['phpipam']['backup_storage_path'],
    backup_keep_days_local:       node['phpipam']['backup_keep_days_local'],
    backup_s3_enabled:            node['phpipam']['backup_s3_enabled'],
    backup_s3_access_key_id:      node['phpipam']['backup_s3_access_key_id'],
    backup_s3_secret_access_key:  node['phpipam']['backup_s3_secret_access_key'],
    backup_s3_region:             node['phpipam']['backup_s3_region'],
    backup_s3_bucket:             node['phpipam']['backup_s3_bucket'],
    backup_s3_path:               node['phpipam']['backup_s3_path'],
    backup_keep_days_s3:          node['phpipam']['backup_keep_days_s3'],
    backup_notifiers:             node['phpipam']['backup_notifiers']
  )
end

cron 'ipam_backup' do
  action :create
  command  "/usr/local/bin/backup perform --config-file #{node['phpipam']['backup_home']}/etc/config.rb --trigger phpipam_backup"
  hour     node['phpipam']['backup_cron_hour']
  minute   node['phpipam']['backup_cron_minute']
  user     node['phpipam']['backup_user']
  weekday  node['phpipam']['backup_cron_weekday']
end


recipes/app_site.rb
=begin
#<
This recipe installs and configures the application and its content. It also
installs Apache 2, if not installed already.
#>
=end

include_recipe 'phpipam::_check_ipam_version'
include_recipe 'apt::default'
include_recipe 'build-essential::default'
include_recipe 'phpipam::_install_apache2'

GIT_SRC = 'https://github.com/phpipam/phpipam.git'.freeze
DEV_SHA256 = '377139d7489376ac3622d936dbce16b1823358a1'.freeze

CONTENT_OWNER = 'www-data'.freeze
CONTENT_GROUP = 'www-data'.freeze

package 'git' do
  action :install
end

directory node['phpipam']['docroot'] do
  action :create
  group CONTENT_GROUP
  recursive true
  user CONTENT_OWNER
end

git 'ipam_download' do
  action :export
  destination node['phpipam']['docroot']
  group CONTENT_GROUP
  repository GIT_SRC
  if node['phpipam']['version'] == 'dev'
    revision DEV_SHA256
  else
    revision node['phpipam']['version']
  end
  user CONTENT_OWNER
end

template "#{node['phpipam']['docroot']}/.htaccess" do
  group CONTENT_GROUP
  mode 0644
  source "htaccess_#{node['phpipam']['version']}.erb"
  user CONTENT_OWNER
  variables(
    uri_base: node['phpipam']['uri_base'],
  )
end

template "#{node['phpipam']['docroot']}/config.php" do
  group CONTENT_GROUP
  mode 0644
  source "config.php_#{node['phpipam']['version']}.erb"
  user CONTENT_OWNER
  variables(
    db_host: node['phpipam']['db_host'],
    db_name: node['phpipam']['db_name'],
    db_password: node['phpipam']['db_password'],
    db_user: node['phpipam']['db_user'],
    uri_base: node['phpipam']['uri_base'],
  )
end


recipes/_install_apache2.rb
include_recipe 'apt::default'

PHP_EXTRA_PACKAGES = [
  'php-pear',
  'php5-cli',
  'php5-curl',
  'php5-gd',
  'php5-gmp',
  'php5-ldap',
  'php5-mcrypt',
  'php5-mysql'
].freeze

PHP5_MANUAL_ENABLE_MODS = %w(
  mcrypt
).freeze

include_recipe "php::#{node['php']['install_method']}"

package 'php-pear' do
  action :install
end

package 'php_extra_packages' do
  package_name PHP_EXTRA_PACKAGES
  action :install
end

PHP5_MANUAL_ENABLE_MODS.each do |phpmod|
  execute "php5enmod #{phpmod}" do
    action :run
    only_if "! php5query -s apache2 #{phpmod}"
    notifies :restart, 'service[apache2]', :delayed
  end
end

node.default['apache']['mpm'] = 'prefork'

web_app 'phpipam' do
  allow_override ['All']
  cookbook 'phpipam'
  docroot node['phpipam']['docroot'].end_with?('/') ? node['phpipam']['docroot'] : "#{node['phpipam']['docroot']}/"
  server_aliases node['phpipam']['vhost_aliases']
  server_name node['phpipam']['vhost_name']
  template 'apache2_vhost.erb'
  uri_base node['phpipam']['uri_base']
end

include_recipe 'apache2::mod_php5'


attributes/default.rb
#<> The host name to connect to for the PHPIPAM database.
default['phpipam']['db_host'] = 'localhost'

#<> The name of the PHPIPAM database.
default['phpipam']['db_name'] = 'phpipam'

#<> The user name for the PHPIPAM database.
default['phpipam']['db_user'] = 'phpipam'

#<> The password for the created PHPIPAM database.
default['phpipam']['db_password'] = 'phpipamadmin'

#<> The root password for the MySQL installation of the PHPIPAM database.
default['phpipam']['db_root_password'] = 'atotallyrandompassword'

#<> The document root, or where to install the PHPIPAM content.
default['phpipam']['docroot'] = '/var/www/html/phpipam'

=begin
#<
Keep the default virtualhost on Apache 2 installations.

You may want to set this to true on a system where PHPIPAM is not the only
site. Leaving this at the default will ensrue that the virtual host installed
is the default virtual host on the system.
#>
=end
default['phpipam']['apache2_keep_default_vhost'] = false

=begin
#<
The URI base to install PHPIPAM to. This will mean that the GUI and API will
be accessible through http(s)://example.com/phpipam/, for example, if this
was set to "/phpipam/".
#>
=end
default['phpipam']['uri_base'] = '/'

#<> The virtual host for the PHPIPAM site.
default['phpipam']['vhost_name'] = nil

#<> Any aliases that should be used to access PHPIPAM.
default['phpipam']['vhost_aliases'] = nil

=begin
#<
The version of IPAM to fetch the content for (and ultimately install).

This can currently be one of `1.2` and `dev`. `dev` is an alias to a current
commit snapshotted off of master, with schema and configuration snapshotted off
of that commit. See `recipes/app_site.rb` for the specific commit dev tracks.
#>
=end
default['phpipam']['version'] = '1.2'

=begin
#<
The location to download the PHPIPAM content to. This is set to a directory
in /tmp to start with, and is overridden to the document root when running
the web content recipes.
#>
=end
default['phpipam']['download_path'] = '/tmp/phpipam'

=begin
#<
The owner of the downloaded content. This is modified to the system's web
user for web installations.
#>
=end
default['phpipam']['content_owner'] = 'root'

=begin
#<
The group of the downloaded content. This is modified to the system's web
group for web installations.
#>
=end
default['phpipam']['content_group'] = 'root'

#<> Install MySQL as part of the cookbook deployment.
default['phpipam']['install_mysql'] = true

=begin
#<
The user to create to perform backups under. A group is also created under
this user name.
#>
=end
default['phpipam']['backup_user'] = 'ipambackup'

#<> The home directory for backup configuration data and content.
default['phpipam']['backup_home'] = '/var/ipambackup'

#<> The directory to hold backup data while performing backups.
default['phpipam']['backup_tmp_path'] = "#{default['phpipam']['backup_home']}/tmp"

=begin
#<
Enable logging backups to a log file. These logs go to the "log" directory
in the directory defined by `node['phpipam']['backup_home']`.
#>
=end
default['phpipam']['backup_enable_logfile'] = true

#<> Enable syslog logging. Logs get logged as `backup` under facility local0.
default['phpipam']['backup_enable_syslog'] = false

#<> The directory to store locally-kept backups.
default['phpipam']['backup_storage_path'] = "#{default['phpipam']['backup_home']}/storage"

#<> Enable S3 backup storage.
default['phpipam']['backup_s3_enabled'] = false

#<> The AWS Access Key ID for S3 backup storage.
default['phpipam']['backup_s3_access_key_id'] = ''

#<> The AWS secret access key for S3 backup storage.
default['phpipam']['backup_s3_secret_access_key'] = ''

#<> The AWS region your bucket is in.
default['phpipam']['backup_s3_region'] = ''

#<> The S3 bucket name for S3 storage.
default['phpipam']['backup_s3_bucket'] = ''

#<> The key (path) S3 backups will be stored in.
default['phpipam']['backup_s3_path'] = ''

#<> The number of days to keep local backups.
default['phpipam']['backup_keep_days_local'] = 7

#<> The number of days to keep S3 backups.
default['phpipam']['backup_keep_days_s3'] = 90

=begin
An array of notifiers. Notifiers follow the general syntax below, based off of
the syntax of the backup gem's [Notifiers resource][1]:

[1]: http://backup.github.io/backup/v4/notifiers/

        default['phpipam']['backup_notifiers'] = [
          {
            type: 'Mail',
            opts: {
              'from' => 'nobody@example.com',
              'to' => 'somebody@example.com',
            }
          }
        ]

Note that this cookbook by default notifies on success, warning, and failure.
=end
default['phpipam']['backup_notifiers'] = []

#<> The minute to run the backup job on (0-59).
default['phpipam']['backup_cron_minute'] = '00'

#<> The hour to run the backup job on (0-23).
default['phpipam']['backup_cron_hour'] =  '02'

#<> The weekday to run the backup job on (0-6, Sun-Sat, or * for every day).
default['phpipam']['backup_cron_weekday'] =  '*'

#<> A list of warning messages to ignore.
default['phpipam']['backup_ignore_warning_messages'] =  []
